src/core/or/circuit_st.h

```diff
@@ -110,6 +110,12 @@ struct circuit_t {
    * circuit-level sendme cells to indicate that we're willing to accept
    * more. */
   int deliver_window;
+  /** FIFO containing the digest of the cells that are just before a SENDME is
+   * sent by the client. It is done at the last cell before our package_window
+   * goes down to 0 which is when we expect a SENDME. The protocol doesn't
+   * allow more than 10 outstanding SENDMEs worth of data meaning this list
+   * should only contain at most 10 digests of 4 bytes each. */
```

Comment:
We should document how these digests correspond to positions in the stream, exactly.  Like, the most recent digest in the list corresponds to the state up until the cell we received at...

Also, "10" and "4" should really be constants here.

Reply:
Attempt in commit a1cdc72082